tinygrad/codegen/uops.py

```diff
@@ -180,6 +180,92 @@ def fix_loop_scope(self, get_recursive_parents:Callable[..., Set[UOp]]):
               break
     self.uops = flatten(loop_stack)
 
+  def replace_op(self, old, new):
+    for v in self.uops: v.vin = tuple(new if x is old else x for x in v.vin)
+    self.uops.remove(old)
+
+  def simplify_phi_loops(self, get_recursive_parents):
+    seen_vars: Dict[str,UOp] = {}
+    def alu_opposite(arg, x, y):
+      if arg == BinaryOps.ADD: return x - y
+      elif arg == BinaryOps.MUL: return Node.__floordiv__(x, y, False)
+      else: raise RuntimeError("unhandled alu")
+    def to_symbolic(u: UOp):
+      if u.uop == UOps.CONST: return NumNode(int(u.arg))
+      elif u.uop in {UOps.LOOP, UOps.SPECIAL}:
+        name = "loop{}".format(len(seen_vars))
+        seen_vars[name] = u
+        return Variable(name, u.vin[0].arg, u.vin[1].arg-1) if u.uop is UOps.LOOP else Variable(name, 0, u.arg[2]-1)
```

Comment:
modified this fn to generate internal loop names so when rendering later, the op can be resolved again